src/stores/notifications/RoomNotificationStateStore.ts

```diff
@@ -96,8 +97,9 @@ export class RoomNotificationStateStore extends AsyncStoreWithClient<IState> {
     private onSync = (state: SyncState, prevState: SyncState | null, data?: ISyncStateData): void => {
         // Only count visible rooms to not torment the user with notification counts in rooms they can't see.
         // This will include highlights from the previous version of the room internally
+        const msc3946ProcessDynamicPredecessor = SettingsStore.getValue("feature_dynamic_room_predecessors");
         const globalState = new SummarizedNotificationState();
-        const visibleRooms = this.matrixClient.getVisibleRooms();
+        const visibleRooms = this.matrixClient.getVisibleRooms(msc3946ProcessDynamicPredecessor);
```

Comment:
This codepath won't be hit if a user is changing the labs flag, as labs are stored in Local Storage and thus won't cause a `/sync` - so it won't update until the next /sync which for smaller accounts may be a little while away.

Reply:
So you think we should listen for changes to this labs flag and re-run this logic when they happen?

Doesn't that seem like overkill? Most users will hit this button once or twice ever, and, if I were hitting it, I don't think I'd expect it to notify me immediately about a room that became visible because I was switching off dynamic room predecessors.

Reply:
It might be overkill but we do so in the other places we check this setting.

![image](https://user-images.githubusercontent.com/2403652/223432825-7307e35a-7bfd-4c4e-b0d5-07233d9c4824.png)
![image](https://user-images.githubusercontent.com/2403652/223432883-79b46bdc-b3df-4fb5-8812-60bdb94ee11e.png)

Reply:
Those places already watched other settings, so the additional complexity was small, and at least in the case of RoomView, it seems useful to re-render them when the labs flag changes.

In RoomNotificationStateStore, we'd be adding the complexity of responding to a settings update while the benefit to the user is questionable. (In fact, I'd argue it would be quite confusing to receive a notification because of changing this labs flag.)

So I do think it's overkill, but happy to discuss further if you disagree.

Reply:
The benefit to the user may be small, but a benefit to the developer is found too. Consistency and predictable data flow.

> (In fact, I'd argue it would be quite confusing to receive a notification because of changing this labs flag.)

But that'd still happen, just on your next /sync so even less predictably.

Reply:
OK, I guess it's better to make it clearer to the user what caused the ping. The benefit to the developer of yet more listeners and events flying around is still definitely arguable but I take your point there too.